src/main/java/ru/ayaz/Chat.java

```diff
@@ -0,0 +1,84 @@
+package ru.ayaz;
+
+
+import ru.ayaz.ru.ayaz.exceptions.InvalidNicknameException;
+import ru.ayaz.ru.ayaz.exceptions.RegistrationFailException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class Chat {
+
+    private ServerSocket serverSocket;
+    private UserMessageDistributor messageDistributor;
+
+    Chat() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    void start() {
+        startUserMessageDistributor();
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                UserSocketHandler userSocketHandler = new UserSocketHandler(socket, messageDistributor);
+                User user = registerUser(userSocketHandler.getWriter(), userSocketHandler.getReader());
```

Comment:
&#x1F34E; **Issue**
Looks to me like the User entity is kind of an inner part of UserSocketHandler and I would even got rid of it and moved the functionality of User class inside of UserSocketHandler.
registerUser function would also be a part of UserSocketHandler - you don't want to have methods like userSocketHandler.getWriter() and userSocketHandler.getReader() on UserSocketHandler. He is supposed to encapsulate them.